### PR TITLE
Add forward and backward support for silu to NestedTensors

### PR DESCRIPTION
**Pull request by drisspg**
# Summary
Add forward and backward support for silu to NestedTensors
- Add forward support to silu
- Add forward support to silu_
- Add backward support to silu
- Add to NT docs
- Add tests

### CODE REVIEW

**Review by cpuhrsch — APPROVED** [review]

File: aten/src/ATen/native/nested/NestedTensorUnaryOps.cpp [hunk 1]
```diff
@@ -76,5 +76,17 @@ Tensor& zero_nested_(Tensor& self) {
...
+  return map_nt(self, at::silu);
+}
+
+Tensor& NestedTensor_silu_(Tensor& self){
+  auto self_ptr = get_nested_tensor_impl(self);
+  check_numel_equals_buffer_size(self_ptr);
+  auto buffer = self_ptr->get_buffer();
+  at::silu_(buffer);
+  return self;
+}
```
Comment (jbschlosser) [inline aten/src/ATen/native/nested/NestedTensorUnaryOps.cpp:89]: nothing to do for this PR, but I wish the in-place variants were more factored out

**Review by jbschlosser — APPROVED** [review]
solid!

File: aten/src/ATen/native/nested/NestedTensorUnaryOps.cpp (hunk 1 above)
Comment (drisspg) [reply aten/src/ATen/native/nested/NestedTensorUnaryOps.cpp:89]: Agreed, I still think we probbably just want to code gen all the things

**Pull request closed without merging**